common/autoware_universe_utils/include/autoware/universe_utils/geometry/geometry.hpp

```diff
@@ -577,6 +577,12 @@ std::optional<geometry_msgs::msg::Point> intersect(
   const geometry_msgs::msg::Point & p1, const geometry_msgs::msg::Point & p2,
   const geometry_msgs::msg::Point & p3, const geometry_msgs::msg::Point & p4);
 
+/**
+ * @brief Check if 2 convex polygons intersect using the GJK algorithm
+ * @details much faster than boost::geometry::intersects()
+ */
+bool intersects(const Polygon2d & convex_polygon1, const Polygon2d & convex_polygon2);
```

Comment:
Should we add the `convex` naming to this function so that developers won't use this function for non-convex polygons?

Reply:
Done :+1: [7037fb1](https://github.com/autowarefoundation/autoware.universe/pull/7853/commits/7037fb1d5852d379377ed33b11b822b21b0f683d)